src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -49,7 +55,7 @@ class ProfilerController
      * @param ContentSecurityPolicyHandler $cspHandler      The Content-Security-Policy handler
      * @param string                       $baseDir         The project root directory
      */
-    public function __construct(UrlGeneratorInterface $generator, Profiler $profiler = null, Environment $twig, array $templates, $toolbarPosition = 'bottom', ContentSecurityPolicyHandler $cspHandler = null, $baseDir = null)
+    public function __construct(UrlGeneratorInterface $generator, Profiler $profiler = null, Environment $twig, array $templates, $toolbarPosition = 'bottom', ContentSecurityPolicyHandler $cspHandler = null, $baseDir = null, KernelInterface $kernel)
```

Comment:
Requiring a kernel makes the controller incompatible with Twig (which does not have one).

And adding a mandatory constructor argument is a BC break.